### PR TITLE
GELU Operator

### PR DESCRIPTION
**Pull request by haojin2**
## Description ##
Address #12984, operator from https://arxiv.org/abs/1606.08415.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] GELU operator (sym/nd.LeakyReLU(act_type='gelu'))
- [x] Unit test
- [x] Gluon interface (gluon.nn.GELU)

### Comments ###
Flakiness also checked:
```
MXNET_TEST_COUNT=10000 nosetests tests/python/unittest/test_operator.py:test_gelu
[INFO] Setting module np/mx/python random seeds, use MXNET_MODULE_SEED=1909868054 to reproduce.
.
----------------------------------------------------------------------
Ran 1 test in 314.370s

OK
MXNET_TEST_COUNT=10000 nosetests tests/python/gpu/test_operator_gpu.py:test_gelu
[INFO] Setting module np/mx/python random seeds, use MXNET_MODULE_SEED=2117426005 to reproduce.
.
----------------------------------------------------------------------
Ran 1 test in 446.883s

OK
```

### CODE REVIEW

**Comment by haojin2** [comment]
@eric-haibin-lin @szha for review

File: tests/python/unittest/test_operator.py [hunk 1]
```diff
@@ -862,6 +862,39 @@ def fselu_grad(grad, x, y):
         check_symbolic_backward(y, [xa], [np.ones(shape)], [ga], rtol=rtol, atol=atol, dtype=dtype)
 
 
+@with_seed()
+def test_gelu():
+    CUBE_CONSTANT = 0.044715
+    TWO_OVER_PI = 0.7978845608028654
+    def g(x):
+        return TWO_OVER_PI * (x + CUBE_CONSTANT * np.power(x, 3))
```
Comment (hendrycks) [inline tests/python/unittest/test_operator.py]: Often x*x*x is cheaper than x ** 3. Also, if one takes inspiration from https://en.wikipedia.org/wiki/Horner's_method , which is often more numerically stable, then g(x) could be

`return TWO_OVER_PI * x * (1 + CUBE_CONSTANT * x * x)`

Comment (haojin2) [reply tests/python/unittest/test_operator.py]: This is not the actual implementation so performance is not critical here... I'll change the internal mshadow math kernel in C++ based on the link you provided.

Comment (haojin2) [reply tests/python/unittest/test_operator.py]: Please check this part for the backend implementation: https://github.com/apache/incubator-mxnet/blob/3bf1279fcd5efb80c0d098583ae213cbd2867b66/src/operator/mshadow_op.h#L134-L147. Thanks!

Comment (hendrycks) [reply tests/python/unittest/test_operator.py]: This looks correct.

Comment (haojin2) [reply tests/python/unittest/test_operator.py]: @hendrycks Thanks for your prompt reply! I'll wait for @eric-haibin-lin and @szha to give a review before we merge this in.

**Comment by karan6181** [comment]
@mxnet-label-bot add [Feature request, Operator, pr-awaiting-review]

File: src/operator/mshadow_op.h [hunk 2]
```diff
@@ -127,6 +131,20 @@ MXNET_UNARY_MATH_OP(softsign, a / (1.0f + math::fabs(a)));
 
 MXNET_UNARY_MATH_OP(softsign_grad, 1.0f /  math::sqr(1.0f + math::fabs(a)));
 
+#define gelu_gx(a) \
```
Comment (eric-haibin-lin) [inline src/operator/mshadow_op.h]: define MXNET_GELU_GX

Comment (haojin2) [reply src/operator/mshadow_op.h]: Done.

**Comment by haojin2** [comment]
@szha Gluon test added, also fixed a problem with existing SELU test.

File: src/operator/mshadow_op.h (hunk 2 above)
Comment (eric-haibin-lin) [reply src/operator/mshadow_op.h]: Can we have MXNET_ as the prefix for the macro? thanks

Comment (haojin2) [reply src/operator/mshadow_op.h]: Done.

**Comment by haojin2** [comment]
@szha Build finally passed, good for merge?

**Pull request merged**